src/engine/bank03.asm

```diff
@@ -3943,7 +3943,169 @@ NPCMovement_dd78 ; dd78 (3:5d78)
 	db EAST
 	db $ff
 
-	INCROM $dd82, $e0b0
+	INCROM $dd82, $dda3
+
+FightingClubAfterDuel: ; dda3 (3:5da3)
+        ld hl, .after_duel_table
+        call FindEndOfBattleScript
+        ret
+; 0xddaa
+
+.after_duel_table
+	db NPC_CHRIS
+	db NPC_CHRIS
+	dw $5e69
+	dw $5e75
+
+	db NPC_MICHAEL
+	db NPC_MICHAEL
+	dw $5e95
+	dw $5ea1
+
+	db NPC_JESSICA
+	db NPC_JESSICA
+	dw $c15e
+	dw $cd5e
+
+	db NPC_MITCH
+	db NPC_MITCH
+	dw Script_BeatMitch
+	dw Script_LoseToMitch
+
+	db $00
+; ddc3
+
+Script_Mitch: ; ddc3 (3:5dc3)
+        start_script
+        run_command ScriptCommand_TryGivePCPack
+        db $02
+        run_command ScriptCommand_JumpIfFlagNonzero2
+        db EVENT_FLAG_0F
+        dw Script_Mitch_AlreadyHaveMedal
+        run_command Func_cf96
```

Comment:
Changed Func_cf96 name. I haven't renamed the events because I'm not sure which event maps to which pupil. It isn't hard to find out but I haven't investigated yet.